src/cloud_provider/kubernetes.rs

```diff
@@ -668,6 +669,19 @@ pub fn delete_environment(
 ) -> Result<(), EngineError> {
     let listeners_helper = ListenersHelper::new(kubernetes.listeners());
 
+    let kubeconfig = kubernetes.get_kubeconfig_file_path()?;
+
+    // check if environment is not already deleted
+    // speed up delete env because of terraform requiring apply + destroy
+    if !kubectl_exec_is_namespace_present(
+        kubeconfig.clone(),
+        environment.namespace(),
+        kubernetes.cloud_provider().credentials_environment_variables(),
+    ) {
+        info!("no need to delete environment {}, already absent", environment.namespace());
```

Comment:
Please avoid using tracing lib when you want to log something on core side and locally with all the context.
Here you've got logger and event_details provided in the function so you can use it:

logger.log(EngineEvent::Info(event_details, EventMessage::from_safe_message("your log"));